src/poprox_recommender/evaluation/generate.py

```diff
@@ -158,32 +163,16 @@ def generate_user_recs(data: Data, pipe_names: list[str] | None = None, n_users:
 
     return user_recs
 
-
-if __name__ == "__main__":
-    """
-    For offline evaluation, set theta in mmr_diversity = 1
-    """
-    print("in poprox-recommender-locality")
-    options = docopt(__doc__)  # type: ignore
-    setup_logging(verbose=options["--verbose"], log_file=options["--log-file"])
-
-    n_users = options["--subset"]
-    if n_users is not None:
-        n_users = int(n_users)
-
-    pipelines = options["--pipelines"]
-    print("Pipelines:", pipelines)
-
-    mind_data = options["--mind-data"]
-    data_path = options["--data_path"]
-    if mind_data is not None:
-        user_recs = generate_user_recs(MindData(mind_data), pipelines, n_users)
-    elif data_path is not None:
-        user_recs = generate_user_recs(PoproxData(data_path), pipelines, n_users)
-
+@hydra.main(config_path="/home/sun00587/research/News_Locality_Polarization/poprox-recommender-locality/src/", config_name="config", version_base="1.1")
+def main(cfg: DictConfig) -> None:
+    user_recs = generate_user_recs(PoproxData(cfg.data_path), cfg.pipelines, 
+                                   theta_topic=cfg.theta_topic, theta_locality=cfg.theta_locality)
+    
+    print(f"Starting run with theta_topic={round(cfg.theta_topic, 2)}, theta_locality={round(cfg.theta_locality, 2)}")
     all_recs = pd.concat(user_recs, ignore_index=True)
-    out_fn = options["--output"]
+    out_fn = "{}top_{}_loc_{}.parquet".format(cfg.output_dir, round(cfg.theta_topic, 2), round(cfg.theta_locality, 2))
```

Comment:
Is there any reason we need to be cross-compatible with the original poprox repo?